toolchain/check/import_ref.cpp

```diff
@@ -371,29 +371,36 @@ class ImportRefResolver {
     return GetLocalConstantId(import_ir_.types().GetInstId(type_id));
   }
 
-  // Returns the ConstantId for a TypeId. Adds unresolved constants to
-  // work_stack_.
-  auto GetLocalCanonicalInstBlockId(SemIR::InstBlockId import_block_id)
-      -> SemIR::InstBlockId {
-    if (!import_block_id.is_valid()) {
-      return SemIR::InstBlockId::Invalid;
+  // Gets the local constant values corresponding to an imported inst block.
+  auto GetLocalInstBlockContents(SemIR::InstBlockId import_block_id)
+      -> llvm::SmallVector<SemIR::InstId> {
+    llvm::SmallVector<SemIR::InstId> inst_ids;
+    if (!import_block_id.is_valid() ||
+        import_block_id == SemIR::InstBlockId::Empty) {
+      return inst_ids;
     }
 
-    auto initial_work = work_stack_.size();
-
     // Import all the values in the block.
     auto import_block = import_ir_.inst_blocks().Get(import_block_id);
-    llvm::SmallVector<SemIR::InstId> inst_ids;
     inst_ids.reserve(import_block.size());
     for (auto import_inst_id : import_block) {
       auto const_id = GetLocalConstantId(import_inst_id);
       inst_ids.push_back(const_id.inst_id());
     }
 
-    if (HasNewWork(initial_work)) {
+    return inst_ids;
+  }
+
+  // Gets a local canonical instruction block ID corresponding to an imported
+  // inst block whose contents were already imported, for example by
+  // GetLocalInstBlockContents.
+  auto GetLocalCanonicalInstBlockId(
+      SemIR::InstBlockId import_block_id,
+      const llvm::SmallVector<SemIR::InstId>& contents) -> SemIR::InstBlockId {
```

Comment:
Should this be arrayref? (only thinking of it because I've seen you do it elsewhere)